contracts/core/chainlink-oracle/v1/ChainlinkOracle.sol

```diff
@@ -7,39 +7,33 @@ import "external-interfaces/IChainlinkAggregatorV3.sol";
 
 import "core/globals/v1/interfaces/IMapleGlobals.sol";
 
-/// @title ChainlinkOracle is a wrapper contract for Chainlink oracle price feeds that allows for manual price feed overrides.
-contract ChainlinkOracle is Ownable {
+import "./interfaces/IChainlinkOracle.sol";
 
-    IChainlinkAggregatorV3 public priceFeed;
-    IMapleGlobals public globals;
+/// @title ChainlinkOracle is a wrapper contract for Chainlink oracle price feeds that allows for manual price feed overrides.
+contract ChainlinkOracle is IChainlinkOracle, Ownable {
 
-    address public immutable assetAddress;
+    IChainlinkAggregatorV3 public override priceFeed;
+    IMapleGlobals public override globals;
 
-    bool   public manualOverride;
-    int256 public manualPrice;
+    address public override immutable assetAddress;
 
-    event ChangeAggregatorFeed(address _newMedianizer, address _oldMedianizer);
-    event       SetManualPrice(int256 _oldPrice, int256 _newPrice);
-    event    SetManualOverride(bool _override);
+    bool   public override manualOverride;
+    int256 public override manualPrice;
 
     /**
         @dev   Creates a new Chainlink based oracle.
         @param _aggregator   Address of Chainlink aggregator.
         @param _assetAddress Address of currency (0x0 for ETH).
         @param _owner        Address of the owner of the contract.
-    */
+     */
     constructor(address _aggregator, address _assetAddress, address _owner) public {
         require(_aggregator != address(0), "CO:ZERO_AGGREGATOR_ADDR");
         priceFeed       = IChainlinkAggregatorV3(_aggregator);
         assetAddress    = _assetAddress;
         transferOwnership(_owner);
     }
 
-    /**
-        @dev    Returns the latest price.
-        @return price The latest price.
-    */
-    function getLatestPrice() public view returns (int256) {
+    function getLatestPrice() public override view returns (int256) {
```

Comment:
Can't we have some CI job that doesn't the Interface verification stuff instead of adding here. It is a pain to keep `override` without any use of it.

Reply:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4d0f8c1da8654a478f046ea7cf83d2166e1025af/contracts/token/ERC20/ERC20.sol#L111

I actually like the `override` keyword, it locks us into the interface function definition.

Reply:
If you define a function with `override` and it does not have a corresponding base function, it will error:
```
contracts/core/loan/v1/Loan.sol:208:38: TypeError: Function has override specified but does not override anything.
    function testFunction() external override {}
```

Reply:
This error also comes up if the function signature changes:
```
contracts/core/loan/v1/Loan.sol:200:53: TypeError: Function has override specified but does not override anything.
    function makeFullPayment(address test) external override {
```

Reply:
Agreed. IMO its a fault of Solidity that one needs to override interfaces at all. The fact that interface prototypes are implicitly `virtual` should mean that implementation functions implicitly override those inherited interface definitions.